### PR TITLE
[MI-3837]: Integrate webhooks and enhancements

### PR DESCRIPTION
**Pull request by SaurabhSharma-884**
- Integrate webhooks to open link modal with slash command
- Integrate webhooks to refresh RHs when linking with slash command
- Update user Avatar component
- Preserve link modal state on retry

### CODE REVIEW

File: webapp/src/components/LinkChannelModal/LinkChannelModal.component.tsx [hunk 1]
```diff
@@ -35,13 +35,16 @@ export const LinkChannelModal = ({onClose}: {onClose: () => void}) => {
     const [mSChannel, setMsChannel] = useState<MSTeamOrChannel | null>(null);
     const [linkChannelsPayload, setLinkChannelsPayload] = useState<LinkChannelsPayload | null>(null);
 
-    const handleModalClose = (preserveFields?: boolean) => {
-        if (!preserveFields) {
+    // console.log(mMChannel, mSChannel, mSTeam);
```
Comment (ayusht2810) [inline webapp/src/components/LinkChannelModal/LinkChannelModal.component.tsx:38]:
```suggestion
```

File: webapp/src/index.tsx [hunk 2]
```diff
@@ -53,12 +54,15 @@ export default class Plugin {
...
+        registry.registerWebSocketEventHandler(`custom_${manifest.id}_link_channels`, handleModalLink(store));
+        registry.registerWebSocketEventHandler(`custom_${manifest.id}_link`, handleLink(store));
     }
 }
 
 declare global {
     interface Window {
-        registerPlugin(id: string, plugin: Plugin): void,
+        registerPlugin(id: string, plugin: Plugin): void;
+        Components: any;
```
Comment (ayusht2810) [inline webapp/src/index.tsx:65]: check if we have type for this

File: yarn.lock
Comment (ayusht2810) [inline yarn.lock:1]: remove this file

**Review by ayusht2810 — APPROVED** [review]

File: webapp/src/index.tsx (hunk 2 above)
Comment (SaurabhSharma-884) [reply webapp/src/index.tsx:65]: https://github.com/mattermost/mattermost/blob/master/webapp/channels/src/plugins/export.js

Components is having type `any` in above file

File: webapp/src/components/LinkChannelModal/LinkChannelModal.component.tsx [hunk 3]
```diff
@@ -35,13 +35,16 @@ export const LinkChannelModal = ({onClose}: {onClose: () => void}) => {
     const [msChannel, setMSChannel] = useState<MSTeamOrChannel | null>(null);
     const [linkChannelsPayload, setLinkChannelsPayload] = useState<LinkChannelsPayload | null>(null);
 
-    const handleModalClose = (preserveFields?: boolean) => {
-        if (!preserveFields) {
+    // console.log(mMChannel, mSChannel, mSTeam);
```
Comment (avas27JTG) [inline webapp/src/components/LinkChannelModal/LinkChannelModal.component.tsx:38]:
```suggestion
```

File: yarn.lock [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
```
Comment (avas27JTG) [inline yarn.lock:4]: I think this was mistakenly generated, remove it.

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**